### PR TITLE
Make StringAssertions.StartWith and String.StartsWith consistent

### PR DESCRIPTION
**Pull request by Serg046**
## IMPORTANT 

* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by a new or existing set of unit tests which follow the Arrange-Act-Assert syntax such as is used [in this example](https://github.com/fluentassertions/fluentassertions/blob/daaf35b9b59b622c96d0c034e8972a020b2bee55/Tests/FluentAssertions.Shared.Specs/BasicEquivalencySpecs.cs#L33).
* [x] If the contribution adds a feature or fixes a bug, please update [**the release notes**](https://github.com/fluentassertions/fluentassertions/tree/master/docs), which are published on the [website](https://fluentassertions.com/releases).
* [x] If the contribution changes the public API the changes needs to be included by running `AcceptApiChanges.ps1`/`AcceptApiChanges.sh`.
* [x] If the contribution affects [the documentation](https://github.com/fluentassertions/fluentassertions/tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).

### CODE REVIEW

**Comment by Serg046** [comment]
Linked issue is #1410.

File: docs/_pages/releases.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ sidebar:
 * Added `ObjectAssertions<TSubject, TAssertions>` to ease creation of custom assertion classes - [#1371](https://github.com/fluentassertions/fluentassertions/pull/1371).
 * Added `ComparingBy{Members,Value}(Type)` to allow specifying open generic types - [#1389](https://github.com/fluentassertions/fluentassertions/pull/1389).
 * Added overload of `CollectionAssertions.NotBeEquivalentTo` that takes a `config` parameter` - [#1408](https://github.com/fluentassertions/fluentassertions/pull/1408).
+* Made StringAssertions.StartWith and String.StartsWith consistent - [#1413](https://github.com/fluentassertions/fluentassertions/pull/1413).
```
Comment (dennisdoomen) [inline docs/_pages/releases.md:18]: Would be nice to rephrase that in such a way that it's clear that this is about empty strings.

**Review by dennisdoomen — APPROVED** [review]

**Review by jnyrup — APPROVED** [review]

File: docs/_pages/releases.md (hunk 1 above)
Comment (Serg046) [reply docs/_pages/releases.md:18]: @dennisdoomen maybe something like this?
Changed `StringAssertions.StartWith`, `StringAssertions.EndWith` and their `EquivalentOf` versions to allow empty strings
Could you advise?

Comment (Serg046) [reply docs/_pages/releases.md]: Changed to that string for now. Feel free to propose another one, I'll commit your one then.

**Comment by jnyrup** [comment]
@Serg046 Thanks for the contribution :+1:

**Pull request merged**